### PR TITLE
e2e testing the use of all props

### LINKED ISSUE #2709: Once assigned a Link label + URL, validation fails if you want to empty the values
If a template has a property type Link, I can safely save if I leave the Link fields blank.
It correctly fails if any of the fields is missing.
It correctly succeeds if I input both fields.

Once I have saved with both fields, if I want to 'delete' that link and return to a 'no value' state, I cannot do it.  Deleting both fields and trying to save returns a validation error.

### LINKED ISSUE #2719: Consistently check zeroing-out values of entity properties
At least Link is failing, could happen to other fields as well.

### PR DESCRIPTION
**Pull request by konzz**
fixes #2719 
fixes #2709 

PR checklist:
- [ ] Update READ.me ?
- [ ] Update API documentation ?

QA checklist:
- [ ] Smoke test the functionality described in the issue
- [ ] Test for side effects
- [ ] UI responsiveness
- [ ] Cross browser testing
- [ ] Code review

### CODE REVIEW

**Comment by konzz** [comment]
blocked by #2580

File: nightmare/paths/metadata.spec.js [hunk 1]
```diff
@@ -11,213 +13,395 @@ describe('metadata path', () => {
...
+        .waitToClick(selectors.settingsView.addNewTemplate)
+        .write(selectors.settingsView.documentTemplateNameForm, 'All props');
+
+      for (let propIndex = 1; propIndex <= 15; propIndex += 1) {
+        await nightmare.waitToClick(localSelectors.propertiesButtons(propIndex));
+      }
+
+      await nightmare
+        .waitToClick(localSelectors.editRelationshipProperty)
+        .select(localSelectors.relationshipPropertyType, '5a8480eac464318833d9b54e')
```
Comment (daneryl) [inline nightmare/paths/metadata.spec.js]: can you use the selectByLabel action you developed on this same pr for all selects ? i thin will help on readability.

File: app/react/Metadata/helpers/validator.js [hunk 2]
```diff
@@ -12,22 +14,70 @@ export function notEmpty(val) {
   return !!val && val.trim() !== '';
 }
 
+export function labelAndUrl(val) {
+  return (
+    !notEmpty(val) ||
+    (notEmpty(val.label) && notEmpty(val.url)) ||
+    (!notEmpty(val.label) && !notEmpty(val.url))
+  );
+}
```
Comment (daneryl) [inline app/react/Metadata/helpers/validator.js]: it seems this is the same function as on the server metadata validation isValidLinkField ? but different  implementations ? we should make an effort to have at least the overlapping validation functions written only once, another example is notEmpty on the client vs isEmpty on the server.

File: nightmare/paths/metadataprops.spec.js [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+    }
+
+    await nightmare
+      .waitToClick(localSelectors.editRelationshipProperty)
+      .selectByLabel(localSelectors.relationshipPropertyType, 'Perpetrator')
+      .waitToClick(selectors.settingsView.saveTemplateButton)
+      .waitToClick('.alert.alert-success');
+  });
+
+  it('should create an entity filling all the props', async () => {
```
Comment (daneryl) [inline nightmare/paths/metadataprops.spec.js]: @konzz i think this one should actually have a expect testing the values are properly saved.

File: nightmare/paths/metadataprops.spec.js [hunk 4]
```diff
@@ -0,0 +1,181 @@
...
+      .type(localSelectors.form.linkUrl, 'https://www.huridocs.org/')
+      .type(localSelectors.form.image, 'test')
+      .type(localSelectors.form.media, 'test')
+      .insert(localSelectors.form.geolocationLat, '46')
+      .insert(localSelectors.form.geolocationLon, '6')
+      .click(localSelectors.form.save)
+      .waitToClick('.alert.alert-success');
+  });
+
+  it('should be able to remove all the values from properties', async () => {
```
Comment (daneryl) [inline nightmare/paths/metadataprops.spec.js]: @konzz i think this one should actually have a expect testing the values are properly deleted.

**Pull request merged**